### PR TITLE
Add typings for `json-ptr@1.2`

### PR DESCRIPTION
**Pull request by treybrisbane**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by treybrisbane** [comment]
I'm assuming I don't need to wrap these in a `declare module 'json-ptr' { ... }`, right? First time submitting. 🙂

File: types/json-ptr/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "baseUrl": "../",
+        "esModuleInterop": true,
```
Comment (sandersn) [inline types/json-ptr/tsconfig.json]: this needs to be false to test backward compatibility

File: types/json-ptr/json-ptr-tests.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import JsonPointer from 'json-ptr';
```
Comment (sandersn) [inline types/json-ptr/json-ptr-tests.ts]: Use the commonjs syntax `import JsonPointer = require('json-ptr')`.

Consumers of the package who want to use `import JsonProinter from 'json-ptr'` will need to add `"esModuleInterop": true`, but Definitely Typed tests need to be backward compatible to show that the types will work with old code bases.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/json-ptr/tsconfig.json (hunk 1 above)
Comment (treybrisbane) [reply types/json-ptr/tsconfig.json]: Done. 🙂

File: types/json-ptr/json-ptr-tests.ts (hunk 2 above)
Comment (treybrisbane) [reply types/json-ptr/json-ptr-tests.ts]: Done. 🙂

**Comment by treybrisbane** [comment]
I've just been informed that the author of `json-ptr` is himself working on a TypeScript reimplementation, which would render these typings redundant.

I'm currently seeking clarification on the ETA of that work. Depending on when it's likely to land, it may be worth merging this anyway during the interim, but it may also be pointless. Can we delay merging this temporarily until I know more? 🙂

**Comment by sandersn** [comment]
Sure, just ping me when you want me to sign off so that the bot will know it's ready to go.

**Comment by treybrisbane** [comment]
Closing this due to the reasons mentioned above. :)

**Pull request closed without merging**